### PR TITLE
US138681 - adding `disallowNull` property

### PR DESCRIPTION
**Pull request by maddiegabriel**
[US138681 - [Consistent Evaluation][Quizzing][UI] Bug fixes with setting null as score for attempt and quiz](https://rally1.rallydev.com/#/42960320374d/dashboard?detail=%2Fuserstory%2F633568672801)

This PR adds a property called `disallowNull` to `d2l-grade-result-numeric-score`. When the component is rendered with `disallow-null`, the red error tooltip (`d2l-validation-custom`) should render on the input box whenever the user clears it.

### CODE REVIEW

File: src/components/d2l-grade-result-numeric-score.js [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { LocalizeMixin } from '@brightspace-ui/core/mixins/localize-mixin.js';
 export class D2LGradeResultNumericScore extends LocalizeMixin(LitElement) {
 	static get properties() {
 		return {
-			scoreNumerator: { type: Number },
+			scoreNumerator: { type: String },
```
Comment (maddiegabriel) [inline src/components/d2l-grade-result-numeric-score.js:11]: Changing type of `scoreNumerator` property from `Number` to `String` 
The reason is that previously, `0` and `undefined` were being treated as the same value (both `0`)
This prevented the component from re-rendering if the user changed the score from a 0 to blank, so the validation error tooltip wasn't showing

File: src/components/d2l-grade-result-numeric-score.js [hunk 2]
```diff
@@ -51,14 +52,14 @@ export class D2LGradeResultNumericScore extends LocalizeMixin(LitElement) {
 
 	render() {
 		let inputNumberLabel;
-		const roundedNumerator = Math.round((this.scoreNumerator + Number.EPSILON) * 100) / 100;
+		const roundedNumerator = Math.round((Number(this.scoreNumerator) + Number.EPSILON) * 100) / 100;
```
Comment (maddiegabriel) [inline src/components/d2l-grade-result-numeric-score.js:55]: Casting `scoreNumerator` to a `Number` in the two places it's used

File: src/components/d2l-grade-result-numeric-score.js [hunk 3]
```diff
@@ -58,7 +62,7 @@ export class D2LGradeResultNumericScore extends LocalizeMixin(LitElement) {
 			inputNumberLabel = this.localize('fullGradeScoreLabel', { numerator: roundedNumerator || 'blank', denominator: this.scoreDenominator });
 		}
 
-		this.isValidScore = !this.validationError || typeof this.validationError === undefined;
```
Comment (maddiegabriel) [inline src/components/d2l-grade-result-numeric-score.js:61]: Moved this check into a function since we do it twice now (`_checkIsValidScore()`)

**Comment by maddiegabriel** [comment]
vdiff changes are due to visual-diff GitHub action upgrade yesteraday: https://d2l.slack.com/archives/C0PHG3QB0/p1652128097236459

**Review by JamieTran — APPROVED** [review]
lgtm

**Review by Scy-D2L — APPROVED** [review]

**Pull request merged**